### ISSUE TITLE
PropertyNotSetInConstructor: false positive with (nested) conditional

### ISSUE THREAD

**Issue by sserbin** [vimeo/psalm#562]
Test case:
```php
<?php
class Foo
{
    /** @var int */
    private $status;

    public function __construct(int $in)
    {
        if (rand(0, 1)) {
            $this->status = 1;
        } else {
            $this->status = $in;
        }
    }
}

```

Expected: no errors
Actual: 
```
Property Foo::$status is not defined in constructor of Foo or in any private methods called in the constructor
    private $status;
```

**Comment by muglug** [comment]
Thanks!
